### DISCUSSION TITLE
CMI Portals Enhancement, Execute CMDs "as console", "as player", "as Op"

### DISCUSSION THREAD

**Discussion by Tarlak333** [Zrips/CMI#5124] [Enhancement]
For the past year the one major issue I have with CMI portals (which I love btw) is it only executes commands "as console" which forces me to frequently use clicking on NPCs to have players execute "as player" or "as Op" commands.  It would be nice if a 3 way toggle could be added to the CMI Portals GUI to set the portal command execution mode to "as console" (CMI) or "as player" (NPCish) or "as Op" (NPCish).  This would most likely involve adding some hidden NPC functionality in the background...but I think its very doable.  This feature would add a tremendous amount of versatility to the already great CMI Portals module.

Thx again.
-Tarlak333

**Comment by Zrips** [comment]
have you tried to use specialized commands? Like simply adding asConsole! or asPlayer! at start of command?
